### PR TITLE
feat(Business Glossary) business glossary and user - tab based profile page

### PR DESCRIPTION
**Pull request by saxo-lalrishav**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

Closing the PR https://github.com/linkedin/datahub/pull/2550 and continuing it from this one.

As per discussion with @gabe-lyons we remove the subview system from glossary and user profile page and add the tab based UI to keep consistency with other entity profile page. Below are the screenshots which shows how our new UI for business glossary and user profile looks like

User Profile Page
image

image

Glossary Profile Page
image

image

### CODE REVIEW

File: datahub-web-react/src/utils/customGraphQL/useGetEntitySearchResults.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { EntityType, SearchInput } from '../../types.generated';
+import { useGetSearchResultsQuery } from '../../graphql/search.generated';
+
+type AllEntityInput<T, K> = Pick<T, Exclude<keyof T, keyof K>> & K;
+
+export function useGetEntitySearchResults(
+    input: AllEntityInput<SearchInput, { type?: EntityType }>,
+    searchTypes: Array<EntityType>,
+) {
+    const result: any = {};
```
Comment (gabe-lyons) [inline datahub-web-react/src/utils/customGraphQL/useGetEntitySearchResults.ts:11]: does this type need to be `any`? Using `any` is going to put callers at risk of throwing javascript errors

Comment (saxo-lalrishav) [reply datahub-web-react/src/utils/customGraphQL/useGetEntitySearchResults.ts:11]: agree, i will fix that

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntityResults.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    flex-grow: 1;
+`;
+
+const DetailWrapper = styled.div`
+    display: inline-flex;
+    width: 100%;
+`;
+
+type Props = {
+    searchResult: { [key in EntityType]?: any[] };
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/entitySearch/RelatedEntityResults.tsx:24]: let's try to remove this `any` as well

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx [hunk 3]
```diff
@@ -24,19 +24,17 @@ const ListItem = styled.div`
...
     if (!entityType) return null;
 
-    const entitiesToShow = ownerships[entityType] || [];
+    const entitiesToShow = searchResult[entityType] || [];
 
     return (
         <ListContainer>
             <TitleContainer>
-                <Typography.Title level={3}>{entityRegistry.getCollectionName(entityType)} owned</Typography.Title>
+                <Typography.Title level={3}>Related {entityRegistry.getCollectionName(entityType)}</Typography.Title>
```
[the remark this answers is not in the document]
Comment (gabe-lyons) [reply datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx:37]: Yep- lets do that.

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntityResults.tsx (hunk 2 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/shared/entitySearch/RelatedEntityResults.tsx]: Done

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx]: Done

File: datahub-web-react/src/utils/customGraphQL/useGetEntitySearchResults.ts (hunk 1 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/utils/customGraphQL/useGetEntitySearchResults.ts]: Done

**Review by gabe-lyons — APPROVED** [review]
thanks- looks good 👍

**Comment by saxo-lalrishav** [comment]
#2142

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**